packages/eslint-plugin/tests/rules/prefer-optional-chain/prefer-optional-chain.test.ts

```diff
@@ -12,6 +12,16 @@ const ruleTester = new RuleTester({
   },
 });
 
+describe('test strict null equality', () => {
+  ruleTester.run('prefer-optional-chain', rule, {
+    valid: [
+      // https://github.com/typescript-eslint/typescript-eslint/issues/7654
+      'data && data.value !== null;',
+    ],
+    invalid: [],
+  });
+});
```

Comment:
In case we want to add more test cases related to this particular case. But I added it to an existing array, as suggested 👍 

changed in commit: https://github.com/typescript-eslint/typescripteslint/pull/8717/commits/f49c9641d12f2f596456789fcd4d27bfc5fac1b3